metricbeat/module/elasticsearch/pending_tasks/pending_tasks.go

```diff
@@ -0,0 +1,77 @@
+package pending_tasks
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/helper"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+	"github.com/elastic/beats/metricbeat/module/elasticsearch"
+)
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	mb.Registry.AddMetricSet("elasticsearch", "pending_tasks", New, hostParser)
+}
+
+var (
+	hostParser = parse.URLHostParserBuilder{
+		DefaultScheme: "http",
+		PathConfigKey: "path",
+		DefaultPath:   "_cluster/pending_tasks",
+	}.Build()
+)
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
```

Comment:
There has been some somplifications to the Elasticsearch metricsets in the last days, see for example https://github.com/elastic/beats/blob/master/metricbeat/module/elasticsearch/index/index.go#L24 We can also address this in a follow up PR. This will work as expected.